File: backend/app/schemas/book_overview.py
from pydantic import BaseModel

from app.schemas.books import BookPayload, GoogleBookDetails
from app.schemas.library import LibraryItemRead
from app.schemas.reviews import UserBookReviewRead


class BookOverview(BaseModel):
    # Aggregates public Google data and private user data for the detail screen.
    book: BookPayload
    google_details: GoogleBookDetails | None = None
    library_item: LibraryItemRead | None = None
    review: UserBookReviewRead | None = None

File: frontend/src/app/models/book-overview.ts
import { Book } from './book';
import { GoogleBookDetails } from './google-book-details';
import { LibraryItem } from './library-item';
import { Review } from './review';

// Detail screen aggregate: public book data plus optional user-specific state.
export interface BookOverview {
  book: Book;
  google_details?: GoogleBookDetails | null;
  library_item?: LibraryItem | null;
  review?: Review | null;
}
